mlrun/projects/project.py

```diff
@@ -2884,6 +2885,69 @@ def _init_function_from_obj(func, project, name=None):
     return name or func.metadata.name, func
 
 
+def _mask_credentials_in_function_object(function):
+    env_to_mask = [
+        "V3IO_ACCESS_KEY",
+        "V3IO_USERNAME",
+        "V3IO_PASSWORD",
+        "V3IO_API",
+        "V3IO_FRAMESD",
+        "AWS_ACCESS_KEY_ID",
+    ]
+    for env in function.spec.env:
+        # if env contain a value and not value_from, this means it wasn't masked then we remove it
+        # if env contain a value_from, this means it was masked then no need to remove it
+        env_name = get_item_name(env, "name")
+        env_value_from = get_item_name(env, "value_from")
+        if env_name in env_to_mask and not env_value_from:
+            set_item_attribute(env, "value", None)
+
+    if (
+        function.metadata.credentials
+        and function.metadata.credentials.access_key
+        and not function.metadata.credentials.access_key.startswith(
+            mlrun.model.Credentials.secret_reference_prefix
+        )
+    ):
+        function.metadata.credentials.access_key = None
+
+    _mask_volumes_in_function_object(function)
+
+    return function
+
+
+def _mask_volumes_in_function_object(function):
+    should_remove_v3io_mounts = False
+    volumes = []
+    for volume in function.spec.volumes:
+        if get_item_name(volume, "name") == "v3io":
+            flex_volume = get_item_name(volume, "flexVolume")
+            if flex_volume:
+                # if it has a secret reference, then it was masked then no need to remove it
+                # just remove the plane access key reference
+                if get_item_name(flex_volume, "secretRef"):
+                    if get_item_name(flex_volume, "options"):
+                        # mainly for sanity that we don't leave any access key not masked
+                        options = get_item_name(flex_volume, "options")
+                        if get_item_name(options, "accessKey"):
+                            set_item_attribute(options, "accessKey", None)
+                # if it doesn't has a secret reference, then it wasn't masked then we remove it
+                else:
+                    should_remove_v3io_mounts = True
+                    continue
+        volumes.append(volume)
+
+    function.spec.volumes = volumes
+    if should_remove_v3io_mounts:
+        volume_mounts = []
+        for volume_mount in function.spec.volume_mounts:
+            if get_item_name(volume_mount, "name") == "v3io":
```

Comment:
Same here - you should probably keep the name of the volume that you found above, and use that instead.